### PR TITLE
Remove sync http handler, not needed anymore since added sync on demand

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/sync/sync_test.go [hunk 1]
```diff
@@ -1144,25 +1089,18 @@ func TestSyncNoImagesByRegex(t *testing.T) {
...
-			resp, err := destClient.R().Post(destBaseURL + "/sync")
-			So(err, ShouldBeNil)
-			So(resp, ShouldNotBeNil)
-			So(resp.StatusCode(), ShouldEqual, 200)
-
-			resp, err = destClient.R().Get(destBaseURL + "/v2/_catalog")
-			So(err, ShouldBeNil)
-			So(resp, ShouldNotBeEmpty)
-			So(resp.StatusCode(), ShouldEqual, 200)
+		resp, err := destClient.R().Get(destBaseURL + "/v2/_catalog")
```
Comment (adodon2go) [inline pkg/extensions/sync/sync_test.go:1092]: I think to simulate better  the test intention we need to add a delay (time.Sleep) before checking that downstream server contains no images. Ideally the upstream server should contain some images to make sure downstream does not synced them.

**Review by adodon2go — COMMENTED** [review]
Overall LGTM

**Review by rchincha — COMMENTED** [review]
LGTM, fix the ci/cd failure and let's merge this.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**